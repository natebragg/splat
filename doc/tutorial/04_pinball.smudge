// 04_pinball.smudge
pinball
{
   * idle (@lockPaddles)
     [
          coin --> prepare-ball
     ],
     prepare-ball (@releaseBall, @resetScore, @unlockPaddles)
     [
          plunger --> run
     ],
     run (@startSound)
     [
          target -(@incScore)-,
          drain -(@sadSound)-> idle
     ] (@displayScore),
}
