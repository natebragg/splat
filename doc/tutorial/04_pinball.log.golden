Locking paddles.
Dropping ball.
Unlocking paddles. Whirrrrrr
Ding ding ding ding ding
DingDing
Awwwwwww
Score: 2
A new high score!
Locking paddles.
Dropping ball.
Unlocking paddles. Whirrrrrr
Ding ding ding ding ding
DingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDingDing
Awwwwwww
Score: 100
A new high score!
Locking paddles.
Dropping ball.
Unlocking paddles. Whirrrrrr
Ding ding ding ding ding
Ding
Awwwwwww
Score: 1
High score: 100
Locking paddles.
